consensus/src/dag/dag_state_sync.rs

```diff
@@ -0,0 +1,180 @@
+use super::{
+    dag_fetcher::{DagFetcher, TDagFetcher},
+    dag_store::Dag,
+    storage::DAGStorage,
+    types::{CertifiedNodeWithLedgerInfo, RemoteFetchRequest},
+    TDAGNetworkSender,
+};
+use crate::state_replication::StateComputer;
+use anyhow::anyhow;
+use aptos_infallible::RwLock;
+use aptos_logger::error;
+use aptos_time_service::TimeService;
+use aptos_types::{
+    epoch_change::EpochChangeProof, epoch_state::EpochState, ledger_info::LedgerInfoWithSignatures,
+};
+use itertools::Itertools;
+use std::sync::Arc;
+
+pub const DAG_WINDOW: u64 = 10;
```

Comment:
we need to move this probably to order rule, since it affects whether a node is included in a commit

Reply:
I was thinking this should be onchain config.